bigbluebutton-html5/imports/ui/components/chat/notification/component.jsx

```diff
@@ -113,13 +113,22 @@ class ChatNotification extends Component {
       disableNotify,
       openChats,
       intl,
+      chatIsOpen,
+      currentChatID,
     } = this.props;
 
+    if (disableNotify) return;
+
+    const hasUnread = unreadCounter => unreadCounter > 0;
+    const isPrivate = id => id !== PUBLIC_KEY;
+    const isNotNotified = (id, unreadCounter) => unreadCounter !== this.state.notified[id];
+    const chatClosed = id => !chatIsOpen || id !== currentChatID;
+
     const chatsNotify = openChats
-      .filter(({ id, unreadCounter }) =>
-        unreadCounter > 0 &&
-        unreadCounter !== this.state.notified[id] &&
-        !disableNotify && id !== PUBLIC_KEY)
+      .filter(({ unreadCounter }) => hasUnread(unreadCounter))
```

Comment:
You could be a little more clever here by doing the destruction on the function signature itself: